sdk/core/System.BinaryData/src/BinaryData.cs

```diff
@@ -0,0 +1,286 @@
+#pragma warning restore AZC0014 // Avoid using banned types in public API
+        {
+            _bytes = JsonSerializer.SerializeToUtf8Bytes(jsonSerializable, type, options);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="BinaryData"/> instance by wrapping the
+        /// provided bytes.
+        /// </summary>
+        /// <param name="data">Byte data to wrap.</param>
+        public BinaryData(ReadOnlyMemory<byte> data)
+        {
+            _bytes = data;
+        }
+
+        /// <summary>
+        /// Creates a <see cref="BinaryData"/> instance from a string by converting
+        /// the string to bytes using the UTF-8 encoding.
+        /// </summary>
+        /// <param name="data">The string data.</param>
+        public BinaryData(string data)
+        {
+            if (data == null)
+            {
+                throw new ArgumentNullException(nameof(data));
+            }
+
+            _bytes = Encoding.UTF8.GetBytes(data);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="BinaryData"/> instance by wrapping the provided
+        /// <see cref="ReadOnlyMemory{Byte}"/>.
+        /// </summary>
+        /// <param name="data">Byte data to wrap.</param>
+        /// <returns>A wrapper over <paramref name="data"/>.</returns>
+        public static BinaryData FromBytes(ReadOnlyMemory<byte> data) =>
+            new BinaryData(data);
+
+        /// <summary>
+        /// Creates a <see cref="BinaryData"/> instance by wrapping the provided
+        /// byte array.
+        /// </summary>
+        /// <param name="data">The array to wrap.</param>
+        /// <returns>A wrapper over <paramref name="data"/>.</returns>
+        public static BinaryData FromBytes(byte[] data) =>
+            new BinaryData(data);
+
+        /// <summary>
+        /// Creates a <see cref="BinaryData"/> instance from a string by converting
+        /// the string to bytes using the UTF-8 encoding.
+        /// </summary>
+        /// <param name="data">The string data.</param>
+        /// <returns>A value representing the UTF-8 encoding of <paramref name="data"/>.</returns>
+        public static BinaryData FromString(string data) =>
+            new BinaryData(data);
+
+        /// <summary>
+        /// Creates a <see cref="BinaryData"/> instance from the specified stream.
+        /// The stream is not disposed by this method.
+        /// </summary>
+        /// <param name="stream">Stream containing the data.</param>
+        /// <returns>A value representing all of the data remaining in <paramref name="stream"/>.</returns>
+        public static BinaryData FromStream(Stream stream)
+        {
+            if (stream == null)
+            {
+                throw new ArgumentNullException(nameof(stream));
+            }
+
+#pragma warning disable AZC0102 // Do not use GetAwaiter().GetResult().
+            return FromStreamAsync(stream, false).GetAwaiter().GetResult();
```

Comment:
Try to avoid sync-over-async code here. Defer instead to the implementation of `Stream.Read`.

Reply:
It isn't sync over async; the boolean parameter controls the helper method as a sync or async implementation.  `FromStreamAsync(stream, false)` always completes synchronously, using synchronous Stream methods.

Reply:
Ok! That's a little counterintuitive from a code review perspective, but if it's a non-user-observable implementation detail then we can always improve it in post as needed. :)